applications/dashboard/controllers/Api/RolesApiController.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+use Garden\Schema\Schema;
+use Garden\Web\Data;
+use Garden\Web\Exception\NotFoundException;
+use Garden\Web\Exception\ServerException;
+use Vanilla\Utility\CapitalCaseScheme;
+
+/**
+ * API Controller for the `/roles` resource.
+ */
+class RolesApiController extends AbstractApiController {
+
+    /** @var CapitalCaseScheme */
+    private $caseScheme;
+
+    /** @var Schema */
+    private $idParamSchema;
+
+    /** @var PermissionModel */
+    private $permissionModel;
+
+    /** @var RoleModel */
+    private $roleModel;
+
+    /** @var Schema */
+    private $rolePostSchema;
+
+    /** @var Schema */
+    private $roleSchema;
+
+    /**
+     * RolesApiController constructor.
+     *
+     * @param RoleModel $roleModel
+     */
+    public function __construct(RoleModel $roleModel, PermissionModel $permissionModel) {
+        $this->roleModel = $roleModel;
+        $this->permissionModel = $permissionModel;
+        $this->caseScheme = new CapitalCaseScheme();
```

Comment:
If memory serves, this was done to make the constructor a little less unwieldy. Injecting `CapitalCaseSchema` didn't seem as beneficial.

```php
CommentsApiController::__construct(CommentModel $commentModel, DiscussionModel $discussionModel, UserModel $userModel, CapitalCaseScheme $caseScheme)
```

becomes

```php
CommentsApiController::__construct(CommentModel $commentModel, DiscussionModel $discussionModel, UserModel $userModel)
```